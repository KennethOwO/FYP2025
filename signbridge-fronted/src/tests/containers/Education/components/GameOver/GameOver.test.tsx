import GameOverPopup from "@root/containers/Education/components/GameOver/GameOver";
import { create } from "react-test-renderer";
import { cleanup } from "@testing-library/react";

jest.mock("react-i18next", () => ({
  useTranslation: () => ({
    t: (key: string) => key, // Mock the translation function to return the key
    i18n: {
      changeLanguage: jest.fn(), // Mock the changeLanguage function
    },
  }),
}));

jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  useNavigate: () => jest.fn(),
}));

describe("Test GameOverPopup", () => {
  afterEach(() => {
    // Clean up the DOM
    cleanup();
    // Reset mocks to their initial state
    jest.resetAllMocks();
  });
  it("should render correctly", () => {
    const props = {
      onClose: jest.fn(),
      score: 2,
    };
    const tree = create(<GameOverPopup {...props} />);
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
