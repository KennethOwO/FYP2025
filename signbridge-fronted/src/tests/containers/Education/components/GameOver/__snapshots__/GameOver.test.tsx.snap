// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test GameOverPopup should render correctly 1`] = `
<div>
  <div>
    <h3>
      game_over
    </h3>
    <div>
      <p>
        your_score
      </p>
      <p>
        2
      </p>
      <button
        onClick={[Function]}
        type="button"
      >
        quit_btn
      </button>
    </div>
  </div>
</div>
`;
